### ISSUE TITLE
LeaderMode is not working properly.

### ISSUE THREAD

**Issue by mkjawadi** [evan-liu/karabiner.ts#153]
The Leader mode is not working properly. It works only once after I execute the writeToProfile and after that everything breaks. Am I missing anything? Below is my hyperLayer config.

```typescript
hyperLayer("spacebar", "hyper-o: open app")
    .leaderMode()
    .notification()
    .manipulators({
      // m: toApp("Mail"),
      c: toApp("Calendar"),
      r: toApp("Reminders"),
      o: toApp("Microsoft Outlook"),
      f: toApp("Finder"),
      a: toApp("Arc"),
      s: toApp("Safari"),
      v: toApp("Visual Studio Code"),
      i: toApp("iTerm"),
      t: toApp("Microsoft Teams"),
      n: toApp("Notes"),
      p: toApp("Passwords"),
    })
    .manipulators([
      mapDoubleTap("m").toApp("Microsoft Outlook").singleTap(toApp("Mail")),
    ])
``` 

Karabiner-Elements v15.3.0
karabiner.ts v1.30.1

**Comment by evan-liu** [comment]
Hi @mkjawadi, does it work without

```ts
    .manipulators([
      mapDoubleTap("m").toApp("Microsoft Outlook").singleTap(toApp("Mail")),
    ])
```

I guess it might be `mapDoubleTap()` not working with `leaderMode()`.

**Comment by mkjawadi** [comment]
Thanks @evan-liu for getting back to me. No, it doesn’t. I’ve tried without that part as well but didn’t work.

**Comment by evan-liu** [comment]
I've tried adding the below to my config, and it worked as expected (I tested with `c` and `s`). There might be conflicts with another part of your config. Can you test with only this layer without any other part of your config? 

```ts
hyperLayer("spacebar", "hyper-o: open app")
    .leaderMode()
    .notification()
    .manipulators({
      // m: toApp("Mail"),
      c: toApp("Calendar"),
      r: toApp("Reminders"),
      o: toApp("Microsoft Outlook"),
      f: toApp("Finder"),
      a: toApp("Arc"),
      s: toApp("Safari"),
      v: toApp("Visual Studio Code"),
      i: toApp("iTerm"),
      t: toApp("Microsoft Teams"),
      n: toApp("Notes"),
      p: toApp("Passwords"),
    }),
```

**Comment by mkjawadi** [comment]
I tried again with just this layer, but the problem persists. As I mentioned before, it works only once launching the first application, and then everything breaks. The 'Hyper + Spacebar' doesn't activate the layer anymore. 🙁

**Comment by evan-liu** [comment]
Hi @mkjawadi, sorry I misunderstood what you mean by "everything breaks". If you only mean "The 'Hyper + Spacebar' doesn't activate the layer anymore" after one action triggered on the layer, that is actually by design. To keep the layer active, you need the [sticky](https://karabiner.ts.evanliu.dev/rules/leader-mode#sticky) option: 

```ts
  hyperLayer("spacebar", "hyper-spacebar: Launch app")
    .leaderMode({ sticky: true })
```

**Comment by evan-liu** [comment]
If you mean the layer cannot be activated again, can you open the Karabiner-EventViewer app, check the events and variables, and see what happens after the first action?

**Comment by evan-liu** [comment]
Sorry, I was testing with an older version. I can reproduce it now after upgrading. Will fix it soon. Sorry about that.

**Comment by evan-liu** [comment]
@mkjawadi Could you please try `1.30.2`.

**Comment by mkjawadi** [comment]
Thanks @evan-liu  for the `1.30.2` release. It resolved the initial problem, but now double tap mappings are not working. I don't know if it's by design? If yes, is there any workaround, as the double tap option is really handy and convenient? This is my config (inspired by your own great [karabiner-config](https://github.com/evan-liu/karabiner-config/blob/main/karabiner-config.ts) 🙏).

```ts
hyperLayer("spacebar", "hyper:spacebar Launch App");
    .leaderMode()
    .notification(hint)
    .manipulators({
     // rest of my config
    })
    .manipulators([
      mapDoubleTap("m").toApp("Microsoft Outlook")
     .singleTap(toApp("Mail")),
      mapDoubleTap("t").toApp("iTerm").singleTap(toApp("Microsoft Teams")),
      mapDoubleTap("v")
        .to(raycastExt("thomas/visual-studio-code/index")) // Raycast: Search recent VS Code Projects
        .singleTap(toApp("Visual Studio Code")),
    ]);
```

**Comment by evan-liu** [comment]
Is double tap not working at all, or only not working on layer leader mode? Can you please open a new issue if it is only not working on layer leader mode? 

I guess the leader mode set the layer var to off on the first tap, so the second tap no longer has the layer condition. If that's the case I will need some time to find a solution.

**Comment by mkjawadi** [comment]
It does work without the `.leaderMode()`. I believe your guess is correct. As requested, I opened a new issue #155.
